elasticdl/python/master/k8s_instance_manager.py

```diff
@@ -232,15 +240,29 @@ def _event_cb(self, event):
                     del self._ps_pod_name_to_id[pod_name]
                     relaunch_ps = self._relaunch_deleted_live_ps
             else:
-                logger.error("Unknown worker pod name: %s" % pod_name)
+                logger.error("Unknown pod name: %s" % pod_name)
                 return
 
-        if relaunch_worker:
+        if relaunch_worker and worker_id:
             logger.info("Relaunching worker.")
-            self._start_worker(self._next_worker_id())
-        elif relaunch_ps:
+            new_worker_id = self._next_worker_id()
+            self._start_worker(new_worker_id)
+            self._update_addr(
+                worker_id,
+                new_worker_id,
+                self._worker_addrs,
+                addr_get_fn=self._k8s_client.get_worker_service_address,
+            )
+        elif relaunch_ps and ps_id:
             logger.info("Relaunching ps.")
-            self._start_ps(ps_id)
+            new_ps_id = self._next_ps_id()
+            self._start_ps(new_ps_id)
+            self._update_addr(
```

Comment:
_update_addr is a static method. We can InstanceManager._update_addr instead of self._update_addr

Reply:
It's only used inside this class now and is marked as internal so I'd like to keep it as `self._update_addr`.